### DISCUSSION TITLE
DNS TTL?

### DISCUSSION THREAD

**Discussion by Ramias1** [tsightler/ring-mqtt#467] [Feature Requests]
Maybe it is a Docker thing, but this container is querying for api.ring.com multiple times per second.  The DNS record for this has at least 200 seconds TTL and requests are served from my Pi Hole DNS cache, but is there a way for the container to respect the TTL and cache locally?  I can see the container does its own DNS at 127.0.0.11 so maybe there is a TTL or DNS cache setting?  Thanks

**Comment by tsightler** [comment]
But isn't Pi Hole already a caching DNS?  In the end Docker just uses the DNS configuration of the host.  NodeJS DNS behavior is to perform a lookup every time.  Ring-mqtt uses ring-client-api, which is uses the got library for HTTP requests so lots of queries sure, but, in general there's an expectation that most people have a local DNS cache so a few lookups a second on the local network should be no big deal.  In the past ring-client-api used cachable-lookup in concert with got, but it just caused too many issues.  Regardless, I don't have any control over ring-client-api, I'm just a consumer.

In the past I had considered including dnsmasq in the container, but honestly, it's just one more thing to maintain and doesn't seem worth it to me.
